Bukkit 1.7.10 and many other outdated Bukkit, Paper versions still seem to be vulnerable by Log4Shell

### Describe the problem

I am using Bukkit version `1.7.10-R0.1-20140808.005431-8` which is the provided version for 1.7.10 available at https://getbukkit.org/download/craftbukkit (which seems to be the provider this image is downloading Bukkit from)

Trying the https://log4shell.huntress.com/ tester results in logged connections by the server and a noticeable delay of the message in chat. This happens even though the server is being run with the `-Dlog4j2.formatMsgNoLookups=true` fallback flag as one can check in the logs.

Trying to mitigate this I then tried using the [CreeperHost/Log4jPatcher](https://github.com/CreeperHost/Log4jPatcher) and this worked flawlessly!
After adding a volume bind and setting `JVM_OPTS=-javaagent:/Log4jPatcher-1.0.1.jar` the tester didn't log any connections by the server and the test message is being sent without any delay.

The patcher also printed this into the logs:
```
2023-04-26T00:15:17.202621677Z + exec mc-server-runner --stop-duration 60s --named-pipe /tmp/minecraft-console-in java -Xmx1G -Xms1G -Dlog4j2.formatMsgNoLookups=true -javaagent:/Log4jPatcher-1.0.1.jar -DIReallyKnowWhatIAmDoingISwear -jar craftbukkit_server-1.7.10-R0.1-20140808.005431-8.jar
2023-04-26T00:15:17.322616373Z Loading libraries, please wait...
2023-04-26T00:15:17.616432106Z [Log4jPatcher] [INFO] Transforming org/apache/logging/log4j/core/lookup/JndiLookup
2023-04-26T00:15:17.739008618Z [Log4jPatcher] [INFO] Transforming org/apache/logging/log4j/core/pattern/MessagePatternConverter
2023-04-26T00:15:17.739784692Z [Log4jPatcher] [WARN]  Unable to find noLookups:Z field in org/apache/logging/log4j/core/pattern/MessagePatternConverter
2023-04-26T00:15:22.329821071Z [00:15:22 INFO]: Starting minecraft server version 1.7.10
2023-04-26T00:15:22.329947396Z [00:15:22 INFO]: Loading properties
```
The warning `Unable to find noLookups:Z field in ...` could also explain why the fallback flag didn't work...

Maybe Log4jPatcher could be included in the image alongside the existing measures to secure affected servers that aren't already covered by the existing flags?

### Container definition

Affected definition:
```yml
version: '3.5'
services:
    minecraft-server:
        image: itzg/minecraft-server:java8-multiarch

        environment:
            - EULA=TRUE
            - TYPE=BUKKIT
            - VERSION=1.7.10-R0.1-20140808.005431-8
            - DEBUG=TRUE

        ports:
            - '25565:25565'
```

Patched using Log4jPatcher:

```yml
version: '3.5'
services:
    minecraft-server:
        image: itzg/minecraft-server:java8-multiarch

        environment:
            - EULA=TRUE
            - TYPE=BUKKIT
            - VERSION=1.7.10-R0.1-20140808.005431-8
            - DEBUG=TRUE
            - JVM_OPTS=-javaagent:/Log4jPatcher-1.0.1.jar

        ports:
            - '25565:25565'

        volumes:
            - './Log4jPatcher-1.0.1.jar:/Log4jPatcher-1.0.1.jar:ro'
```

### Container logs

```
2023-04-25T23:19:46.128590358Z + export HOME=/data
2023-04-25T23:19:46.128645350Z + HOME=/data
2023-04-25T23:19:46.128935721Z ++ id -u
2023-04-25T23:19:46.130440793Z ++ id -g
2023-04-25T23:19:46.131635507Z ++ ls -lnd /data
2023-04-25T23:19:46.133375445Z + log 'Running as uid=1000 gid=1000 with /data as '\''drwxrwxr-x 2 1000 1000 4096 Apr 19 13:19 /data'\'''
2023-04-25T23:19:46.133388019Z + local oldState
2023-04-25T23:19:46.133702485Z ++ shopt -po xtrace
2023-04-25T23:19:46.134046315Z + oldState='set -o xtrace'
2023-04-25T23:19:46.134137244Z + shopt -u -o xtrace
2023-04-25T23:19:46.135626146Z [init] 2023-04-25 23:19:46+00:00 Running as uid=1000 gid=1000 with /data as 'drwxrwxr-x 2 1000 1000 4096 Apr 19 13:19 /data'
2023-04-25T23:19:46.135713618Z + '[' '!' -e /data/eula.txt ']'
2023-04-25T23:19:46.135719409Z + isTrue TRUE
2023-04-25T23:19:46.135746951Z + case "${1,,}" in
2023-04-25T23:19:46.135750698Z + return 0
2023-04-25T23:19:46.135753272Z + writeEula
2023-04-25T23:19:46.136210854Z ++ date
2023-04-25T23:19:46.136948257Z + echo '# Generated via Docker
2023-04-25T23:19:46.136958075Z # Tue 25 Apr 2023 11:19:46 PM UTC
2023-04-25T23:19:46.136961011Z eula=true
2023-04-25T23:19:46.137027565Z '
2023-04-25T23:19:46.137230132Z + isTrue false
2023-04-25T23:19:46.137312816Z + case "${1,,}" in
2023-04-25T23:19:46.137393857Z + return 1
2023-04-25T23:19:46.137539548Z + isTrue false
2023-04-25T23:19:46.137619517Z + case "${1,,}" in
2023-04-25T23:19:46.137698393Z + return 1
2023-04-25T23:19:46.137838935Z + [[ -n '' ]]
2023-04-25T23:19:46.137994184Z + [[ -n '' ]]
2023-04-25T23:19:46.138131489Z + echo password=8952634760c6704c221eedea
2023-04-25T23:19:46.138366487Z + fixJavaPath
2023-04-25T23:19:46.138475981Z + which java
2023-04-25T23:19:46.139779187Z + export VERSIONS_JSON=https://launchermeta.mojang.com/mc/game/version_manifest.json
2023-04-25T23:19:46.139868964Z + VERSIONS_JSON=https://launchermeta.mojang.com/mc/game/version_manifest.json
2023-04-25T23:19:46.140016770Z + case "X$VERSION" in
2023-04-25T23:19:46.140316157Z + VANILLA_VERSION=1.7.10-R0.1-20140808.005431-8
2023-04-25T23:19:46.140464863Z + export VANILLA_VERSION
2023-04-25T23:19:46.140900244Z ++ get_major_version 1.7.10-R0.1-20140808.005431-8
2023-04-25T23:19:46.140912908Z ++ version=1.7.10-R0.1-20140808.005431-8
2023-04-25T23:19:46.141186367Z ++ echo 1.7.10-R0.1-20140808.005431-8
2023-04-25T23:19:46.141429029Z ++ cut -d. -f 1-2
2023-04-25T23:19:46.142420164Z + MAJOR_VANILLA_VERSION=1.7
2023-04-25T23:19:46.142565655Z + export MAJOR_VANILLA_VERSION
2023-04-25T23:19:46.142711446Z + log 'Resolved version given 1.7.10-R0.1-20140808.005431-8 into 1.7.10-R0.1-20140808.005431-8 and major version 1.7'
2023-04-25T23:19:46.142837781Z + local oldState
2023-04-25T23:19:46.143247454Z ++ shopt -po xtrace
2023-04-25T23:19:46.143618505Z + oldState='set -o xtrace'
2023-04-25T23:19:46.143628894Z + shopt -u -o xtrace
2023-04-25T23:19:46.144888238Z [init] 2023-04-25 23:19:46+00:00 Resolved version given 1.7.10-R0.1-20140808.005431-8 into 1.7.10-R0.1-20140808.005431-8 and major version 1.7
2023-04-25T23:19:46.144978707Z + cd /data
2023-04-25T23:19:46.144987673Z + export ORIGINAL_TYPE=BUKKIT
2023-04-25T23:19:46.144991861Z + ORIGINAL_TYPE=BUKKIT
2023-04-25T23:19:46.144995467Z + isTrue false
2023-04-25T23:19:46.144999064Z + case "${1,,}" in
2023-04-25T23:19:46.145007750Z + return 1
2023-04-25T23:19:46.145011918Z + isTrue false
2023-04-25T23:19:46.145015565Z + case "${1,,}" in
2023-04-25T23:19:46.145047134Z + return 1
2023-04-25T23:19:46.145098600Z + [[ -n '' ]]
2023-04-25T23:19:46.145105643Z + [[ -n '' ]]
2023-04-25T23:19:46.145109080Z + [[ -n '' ]]
2023-04-25T23:19:46.145112576Z + [[ -n '' ]]
2023-04-25T23:19:46.145119980Z + [[ -n '' ]]
2023-04-25T23:19:46.145151428Z + versionLessThan 1.7
2023-04-25T23:19:46.145162088Z + mc-image-helper compare-versions 1.7.10-R0.1-20140808.005431-8 lt 1.7
2023-04-25T23:19:47.036595564Z + return 1
2023-04-25T23:19:47.036630979Z + log 'Resolving type given BUKKIT'
2023-04-25T23:19:47.036641940Z + local oldState
2023-04-25T23:19:47.037137673Z ++ shopt -po xtrace
2023-04-25T23:19:47.037465883Z + oldState='set -o xtrace'
2023-04-25T23:19:47.037478086Z + shopt -u -o xtrace
2023-04-25T23:19:47.038994880Z [init] 2023-04-25 23:19:47+00:00 Resolving type given BUKKIT
2023-04-25T23:19:47.039321258Z + case "${TYPE^^}" in
2023-04-25T23:19:47.039330695Z + exec /start-deployBukkitSpigot
2023-04-25T23:19:47.042141878Z + set -e
2023-04-25T23:19:47.042646027Z + isTrue ''
2023-04-25T23:19:47.042658861Z + case "${1,,}" in
2023-04-25T23:19:47.042662448Z + return 1
2023-04-25T23:19:47.042665624Z + isTrue ''
2023-04-25T23:19:47.042669020Z + case "${1,,}" in
2023-04-25T23:19:47.042701861Z + return 1
2023-04-25T23:19:47.042730745Z + downloadSpigot
2023-04-25T23:19:47.042813730Z + local match
2023-04-25T23:19:47.042858192Z + case "$TYPE" in
2023-04-25T23:19:47.042912172Z + match=CraftBukkit
2023-04-25T23:19:47.042961825Z + downloadUrl=
2023-04-25T23:19:47.042987322Z + getbukkitFlavor=craftbukkit
2023-04-25T23:19:47.043048948Z + [[ 1.7.10-R0.1-20140808.005431-8 = LATEST ]]
2023-04-25T23:19:47.043054016Z + [[ -z '' ]]
2023-04-25T23:19:47.043056611Z + versionLessThan 1.16.5
2023-04-25T23:19:47.043171636Z + mc-image-helper compare-versions 1.7.10-R0.1-20140808.005431-8 lt 1.16.5
2023-04-25T23:19:47.893455055Z + return 0
2023-04-25T23:19:47.893487846Z + downloadUrl=https://cdn.getbukkit.org/craftbukkit/craftbukkit-1.7.10-R0.1-20140808.005431-8.jar
2023-04-25T23:19:47.893498446Z + setServerVar
2023-04-25T23:19:47.893506881Z + case "$TYPE" in
2023-04-25T23:19:47.893511159Z + export SERVER=craftbukkit_server-1.7.10-R0.1-20140808.005431-8.jar
2023-04-25T23:19:47.893515117Z + SERVER=craftbukkit_server-1.7.10-R0.1-20140808.005431-8.jar
2023-04-25T23:19:47.893614070Z + '[' -f craftbukkit_server-1.7.10-R0.1-20140808.005431-8.jar ']'
2023-04-25T23:19:47.893626784Z + isDebugging
2023-04-25T23:19:47.893643295Z + isTrue TRUE
2023-04-25T23:19:47.893648074Z + case "${1,,}" in
2023-04-25T23:19:47.893651961Z + return 0
2023-04-25T23:19:47.893655729Z + return 0
2023-04-25T23:19:47.893749593Z + curlArgs=' -v'
2023-04-25T23:19:47.893760673Z + log 'Downloading CraftBukkit from https://cdn.getbukkit.org/craftbukkit/craftbukkit-1.7.10-R0.1-20140808.005431-8.jar ...'
2023-04-25T23:19:47.893764460Z + local oldState
2023-04-25T23:19:47.894345583Z ++ shopt -po xtrace
2023-04-25T23:19:47.894631916Z + oldState='set -o xtrace'
2023-04-25T23:19:47.894660408Z + shopt -u -o xtrace
2023-04-25T23:19:47.896244798Z [init] 2023-04-25 23:19:47+00:00 Downloading CraftBukkit from https://cdn.getbukkit.org/craftbukkit/craftbukkit-1.7.10-R0.1-20140808.005431-8.jar ...
2023-04-25T23:19:47.896577597Z + curl -fsSL -o craftbukkit_server-1.7.10-R0.1-20140808.005431-8.jar -v https://cdn.getbukkit.org/craftbukkit/craftbukkit-1.7.10-R0.1-20140808.005431-8.jar
2023-04-25T23:19:47.915292799Z *   Trying 149.28.57.95:443...
2023-04-25T23:19:47.915376936Z * TCP_NODELAY set
2023-04-25T23:19:47.997411975Z * Connected to cdn.getbukkit.org (149.28.57.95) port 443 (#0)
2023-04-25T23:19:47.997814595Z * ALPN, offering h2
2023-04-25T23:19:47.997821307Z * ALPN, offering http/1.1
2023-04-25T23:19:48.004561094Z * successfully set certificate verify locations:
2023-04-25T23:19:48.004591200Z *   CAfile: /etc/ssl/certs/ca-certificates.crt
2023-04-25T23:19:48.004595267Z   CApath: /etc/ssl/certs
2023-04-25T23:19:48.005127038Z } [5 bytes data]
2023-04-25T23:19:48.005139741Z * TLSv1.3 (OUT), TLS handshake, Client hello (1):
2023-04-25T23:19:48.005143558Z } [512 bytes data]
2023-04-25T23:19:48.088165956Z * TLSv1.3 (IN), TLS handshake, Server hello (2):
2023-04-25T23:19:48.088208054Z { [98 bytes data]
2023-04-25T23:19:48.088211952Z * TLSv1.2 (IN), TLS handshake, Certificate (11):
2023-04-25T23:19:48.088215117Z { [4028 bytes data]
2023-04-25T23:19:48.088806088Z * TLSv1.2 (IN), TLS handshake, Server key exchange (12):
2023-04-25T23:19:48.088819081Z { [333 bytes data]
2023-04-25T23:19:48.088881038Z * TLSv1.2 (IN), TLS handshake, Server finished (14):
2023-04-25T23:19:48.088887950Z { [4 bytes data]
2023-04-25T23:19:48.089255584Z * TLSv1.2 (OUT), TLS handshake, Client key exchange (16):
2023-04-25T23:19:48.089263609Z } [70 bytes data]
2023-04-25T23:19:48.089313642Z * TLSv1.2 (OUT), TLS change cipher, Change cipher spec (1):
2023-04-25T23:19:48.089319513Z } [1 bytes data]
2023-04-25T23:19:48.089442162Z * TLSv1.2 (OUT), TLS handshake, Finished (20):
2023-04-25T23:19:48.089449725Z } [16 bytes data]
2023-04-25T23:19:48.171322593Z * TLSv1.2 (IN), TLS handshake, Finished (20):
2023-04-25T23:19:48.171366524Z { [16 bytes data]
2023-04-25T23:19:48.171374629Z * SSL connection using TLSv1.2 / ECDHE-RSA-AES256-GCM-SHA384
2023-04-25T23:19:48.171378657Z * ALPN, server accepted to use h2
2023-04-25T23:19:48.171382093Z * Server certificate:
2023-04-25T23:19:48.171385259Z *  subject: CN=cdn.getbukkit.org
2023-04-25T23:19:48.171388375Z *  start date: Mar 16 15:31:37 2023 GMT
2023-04-25T23:19:48.171391811Z *  expire date: Jun 14 15:31:36 2023 GMT
2023-04-25T23:19:48.171404394Z *  subjectAltName: host "cdn.getbukkit.org" matched cert's "cdn.getbukkit.org"
2023-04-25T23:19:48.171455790Z *  issuer: C=US; O=Let's Encrypt; CN=R3
2023-04-25T23:19:48.171627179Z *  SSL certificate verify ok.
2023-04-25T23:19:48.171810471Z * Using HTTP2, server supports multi-use
2023-04-25T23:19:48.171819798Z * Connection state changed (HTTP/2 confirmed)
2023-04-25T23:19:48.171870312Z * Copying HTTP/2 data in stream buffer to connection buffer after upgrade: len=0
2023-04-25T23:19:48.171874840Z } [5 bytes data]
2023-04-25T23:19:48.172048253Z * Using Stream ID: 1 (easy handle 0x55e9a1433300)
2023-04-25T23:19:48.172101152Z } [5 bytes data]
2023-04-25T23:19:48.172246483Z > GET /craftbukkit/craftbukkit-1.7.10-R0.1-20140808.005431-8.jar HTTP/2
2023-04-25T23:19:48.172255759Z > Host: cdn.getbukkit.org
2023-04-25T23:19:48.172259496Z > user-agent: curl/7.68.0
2023-04-25T23:19:48.172263053Z > accept: */*
2023-04-25T23:19:48.172266619Z > 
2023-04-25T23:19:48.172270096Z { [5 bytes data]
2023-04-25T23:19:48.172272681Z * Connection state changed (MAX_CONCURRENT_STREAMS == 128)!
2023-04-25T23:19:48.172290674Z } [5 bytes data]
2023-04-25T23:19:48.253688929Z < HTTP/2 200 
2023-04-25T23:19:48.253735015Z < server: nginx/1.10.3 (Ubuntu)
2023-04-25T23:19:48.253743932Z < date: Tue, 25 Apr 2023 23:19:48 GMT
2023-04-25T23:19:48.253748310Z < content-type: application/java-archive
2023-04-25T23:19:48.253757597Z < content-length: 20471204
2023-04-25T23:19:48.253767476Z < last-modified: Mon, 20 Feb 2017 10:06:34 GMT
2023-04-25T23:19:48.253771994Z < etag: "58aabfaa-1385da4"
2023-04-25T23:19:48.253778907Z < strict-transport-security: max-age=15768000; preload; always;
2023-04-25T23:19:48.253788395Z < x-content-type-options: nosniff
2023-04-25T23:19:48.253795318Z < x-xss-protection: 1; mode=block
2023-04-25T23:19:48.253799124Z < x-robots-tag: none
2023-04-25T23:19:48.253807229Z < content-security-policy: frame-ancestors 'self'
2023-04-25T23:19:48.253828680Z < x-frame-options: SAMEORIGIN
2023-04-25T23:19:48.253839189Z < x-server-powered-by: Vyzra.com
2023-04-25T23:19:48.253846593Z < accept-ranges: bytes
2023-04-25T23:19:48.253850060Z < 
2023-04-25T23:19:48.254079086Z { [5 bytes data]
2023-04-25T23:19:50.972220599Z * Connection #0 to host cdn.getbukkit.org left intact
2023-04-25T23:19:50.976283251Z + [[ 0 != 0 ]]
2023-04-25T23:19:50.976873029Z ++ grep -c 'DOCTYPE html' craftbukkit_server-1.7.10-R0.1-20140808.005431-8.jar
2023-04-25T23:19:50.991231760Z + [[ 0 != 0 ]]
2023-04-25T23:19:50.991263018Z + JVM_OPTS=' -DIReallyKnowWhatIAmDoingISwear'
2023-04-25T23:19:50.991266315Z + export JVM_OPTS
2023-04-25T23:19:50.991409321Z + export FAMILY=SPIGOT
2023-04-25T23:19:50.991467840Z + FAMILY=SPIGOT
2023-04-25T23:19:50.991471967Z + exec /start-spiget
2023-04-25T23:19:50.994598017Z + : ''
2023-04-25T23:19:50.994639223Z + : 5
2023-04-25T23:19:50.994643902Z + : '*.jar,*-version.json'
2023-04-25T23:19:50.994647649Z + acceptArgs=(--accept application/zip --accept application/java-archive --accept application/octet-stream)
2023-04-25T23:19:50.994918002Z + [[ -n '' ]]
2023-04-25T23:19:50.994932740Z + exec /start-setupWorld
2023-04-25T23:19:51.000787108Z + '[' BUKKIT = CURSEFORGE ']'
2023-04-25T23:19:51.000853762Z + worldDest=/data/world
2023-04-25T23:19:51.000858371Z + [[ -n '' ]]
2023-04-25T23:19:51.000861837Z + exec /start-setupVanillaTweaks
2023-04-25T23:19:51.003475724Z + out_dir=/data/world/datapacks
2023-04-25T23:19:51.003523302Z + isTrue false
2023-04-25T23:19:51.003527350Z + case "${1,,}" in
2023-04-25T23:19:51.003573676Z + return 1
2023-04-25T23:19:51.003596228Z + [[ -n '' ]]
2023-04-25T23:19:51.003610315Z + [[ -n '' ]]
2023-04-25T23:19:51.003613491Z + exec /start-setupForgeApiMods
2023-04-25T23:19:51.011206856Z + isTrue false
2023-04-25T23:19:51.011251800Z + case "${1,,}" in
2023-04-25T23:19:51.011256739Z + return 1
2023-04-25T23:19:51.011260446Z + '[' '' ']'
2023-04-25T23:19:51.011264033Z + '[' '' ']'
2023-04-25T23:19:51.011267670Z + exec /start-setupModpack
2023-04-25T23:19:51.011271367Z + CURSE_URL_BASE=https://minecraft.curseforge.com/projects
2023-04-25T23:19:51.011275353Z + isTrue false
2023-04-25T23:19:51.011278940Z + case "${1,,}" in
2023-04-25T23:19:51.011282857Z + return 1
2023-04-25T23:19:51.011506724Z + handlePackwiz
2023-04-25T23:19:51.011518496Z + [[ -n '' ]]
2023-04-25T23:19:51.011522163Z + handleModpackListOrFile
2023-04-25T23:19:51.011525469Z + [[ -n '' ]]
2023-04-25T23:19:51.011529347Z + [[ -n '' ]]
2023-04-25T23:19:51.011532933Z + [[ -n '' ]]
2023-04-25T23:19:51.011786555Z + handleCurseForgeManifest
2023-04-25T23:19:51.011795001Z + [[ -n '' ]]
2023-04-25T23:19:51.011798758Z + handleGenericPacks
2023-04-25T23:19:51.011802525Z + : ''
2023-04-25T23:19:51.011806623Z + : ''
2023-04-25T23:19:51.011810600Z + : ''
2023-04-25T23:19:51.011814287Z + [[ -n '' ]]
2023-04-25T23:19:51.011817804Z + handleModrinthProjects
2023-04-25T23:19:51.011821401Z + : ''
2023-04-25T23:19:51.011824787Z + : true
2023-04-25T23:19:51.011840185Z + : release
2023-04-25T23:19:51.011844002Z + [[ -n '' ]]
2023-04-25T23:19:51.011847478Z + exec /start-setupModconfig
2023-04-25T23:19:51.016987999Z + isTrue true
2023-04-25T23:19:51.017030828Z + case "${1,,}" in
2023-04-25T23:19:51.017034816Z + return 0
2023-04-25T23:19:51.017038162Z + updateArg=--skip-newer-in-destination
2023-04-25T23:19:51.017041508Z + isTrue true
2023-04-25T23:19:51.017045275Z + case "${1,,}" in
2023-04-25T23:19:51.017048271Z + return 0
2023-04-25T23:19:51.017051317Z + subcommand=sync-and-interpolate
2023-04-25T23:19:51.017072236Z + : /plugins
2023-04-25T23:19:51.017077115Z + : /data/plugins
2023-04-25T23:19:51.017080631Z + '[' -d /plugins ']'
2023-04-25T23:19:51.017155030Z + : /mods
2023-04-25T23:19:51.017163356Z + : /data/mods
2023-04-25T23:19:51.017183983Z + '[' -d /mods ']'
2023-04-25T23:19:51.017187400Z + : /config
2023-04-25T23:19:51.017203290Z + : /data/config
2023-04-25T23:19:51.017255226Z + '[' -d /config ']'
2023-04-25T23:19:51.017262039Z + exec /start-setupServerProperties
2023-04-25T23:19:51.023934170Z [init] 2023-04-25 23:19:51+00:00 Creating server properties in /data/server.properties
2023-04-25T23:19:51.027523672Z [init] 2023-04-25 23:19:51+00:00 Disabling whitelist functionality
2023-04-25T23:19:51.033281110Z [init] 2023-04-25 23:19:51+00:00 DEBUG: Adding white-list with 'false' in /data/server.properties
2023-04-25T23:19:51.039330442Z [init] 2023-04-25 23:19:51+00:00 DEBUG: Adding enable-rcon with 'true' in /data/server.properties
2023-04-25T23:19:51.047288347Z [init] 2023-04-25 23:19:51+00:00 DEBUG: Adding rcon.password with '*****' in /data/server.properties
2023-04-25T23:19:51.051137342Z [init] 2023-04-25 23:19:51+00:00 DEBUG: Adding rcon.port with '25575' in /data/server.properties
2023-04-25T23:19:52.033136087Z [init] 2023-04-25 23:19:52+00:00 DEBUG: Adding motd with 'A Bukkit Minecraft Server powered by Docker' in /data/server.properties
2023-04-25T23:19:52.039520823Z + versionLessThan 1.7.6
2023-04-25T23:19:52.039557181Z + mc-image-helper compare-versions 1.7.10-R0.1-20140808.005431-8 lt 1.7.6
2023-04-25T23:19:52.954006384Z + return 1
2023-04-25T23:19:52.954410997Z + opsFile=ops.json
2023-04-25T23:19:52.954531561Z + whitelistFile=whitelist.json
2023-04-25T23:19:52.954914935Z + isTrue ''
2023-04-25T23:19:52.955044586Z + case "${1,,}" in
2023-04-25T23:19:52.955143671Z + return 1
2023-04-25T23:19:52.955277490Z + '[' -n '' ']'
2023-04-25T23:19:52.955408334Z + '[' -n '' ']'
2023-04-25T23:19:52.955525080Z + isTrue ''
2023-04-25T23:19:52.955608726Z + case "${1,,}" in
2023-04-25T23:19:52.955682394Z + return 1
2023-04-25T23:19:52.955814349Z + '[' -n '' ']'
2023-04-25T23:19:52.955932610Z + '[' -n '' ']'
2023-04-25T23:19:52.956053074Z + exec /start-finalExec
2023-04-25T23:19:52.958839991Z + '[' -n '' ']'
2023-04-25T23:19:52.958964123Z + canUseRollingLogs=true
2023-04-25T23:19:52.959047568Z + useFallbackJvmFlag=false
2023-04-25T23:19:52.959153646Z + versionLessThan 1.7
2023-04-25T23:19:52.959193570Z + mc-image-helper compare-versions 1.7.10-R0.1-20140808.005431-8 lt 1.7
2023-04-25T23:19:53.813597850Z + return 1
2023-04-25T23:19:53.813626845Z + isFamily VANILLA
2023-04-25T23:19:53.813629971Z + for f in "${@}"
2023-04-25T23:19:53.813632726Z + [[ SPIGOT == \V\A\N\I\L\L\A ]]
2023-04-25T23:19:53.813635621Z + return 1
2023-04-25T23:19:53.813678721Z + isFamily VANILLA
2023-04-25T23:19:53.813709028Z + for f in "${@}"
2023-04-25T23:19:53.813712754Z + [[ SPIGOT == \V\A\N\I\L\L\A ]]
2023-04-25T23:19:53.813715599Z + return 1
2023-04-25T23:19:53.813723845Z + isType PURPUR
2023-04-25T23:19:53.813726460Z + for t in "${@}"
2023-04-25T23:19:53.813729085Z + [[ BUKKIT == \P\U\R\P\U\R ]]
2023-04-25T23:19:53.813731790Z + return 1
2023-04-25T23:19:53.813738833Z + isType PURPUR
2023-04-25T23:19:53.813741358Z + for t in "${@}"
2023-04-25T23:19:53.813743852Z + [[ BUKKIT == \P\U\R\P\U\R ]]
2023-04-25T23:19:53.813746527Z + return 1
2023-04-25T23:19:53.813749092Z + versionLessThan 1.18.1
2023-04-25T23:19:53.813795358Z + mc-image-helper compare-versions 1.7.10-R0.1-20140808.005431-8 lt 1.18.1
2023-04-25T23:19:54.774583257Z + return 0
2023-04-25T23:19:54.774643529Z + useFallbackJvmFlag=true
2023-04-25T23:19:54.774788208Z + true
2023-04-25T23:19:54.774807014Z + JVM_OPTS='-Dlog4j2.formatMsgNoLookups=true  -DIReallyKnowWhatIAmDoingISwear'
2023-04-25T23:19:54.774968333Z + isTrue false
2023-04-25T23:19:54.774981468Z + case "${1,,}" in
2023-04-25T23:19:54.775041990Z + return 1
2023-04-25T23:19:54.775150813Z + versionLessThan 1.14
2023-04-25T23:19:54.775190497Z + mc-image-helper compare-versions 1.7.10-R0.1-20140808.005431-8 lt 1.14
2023-04-25T23:19:55.694993116Z + return 0
2023-04-25T23:19:55.695033932Z + [[ '' = false ]]
2023-04-25T23:19:55.695093031Z + [[ '' = false ]]
2023-04-25T23:19:55.695170225Z + expandedDOpts=
2023-04-25T23:19:55.695256105Z + '[' -n '' ']'
2023-04-25T23:19:55.695348117Z + isTrue ''
2023-04-25T23:19:55.695426723Z + case "${1,,}" in
2023-04-25T23:19:55.695470134Z + return 1
2023-04-25T23:19:55.695650028Z + isTrue ''
2023-04-25T23:19:55.695717183Z + case "${1,,}" in
2023-04-25T23:19:55.695747240Z + return 1
2023-04-25T23:19:55.695819354Z + isTrue ''
2023-04-25T23:19:55.695842958Z + case "${1,,}" in
2023-04-25T23:19:55.695884645Z + return 1
2023-04-25T23:19:55.695966358Z + isTrue ''
2023-04-25T23:19:55.695998306Z + case "${1,,}" in
2023-04-25T23:19:55.696040215Z + return 1
2023-04-25T23:19:55.696131766Z + isTrue ''
2023-04-25T23:19:55.696162172Z + case "${1,,}" in
2023-04-25T23:19:55.696219078Z + return 1
2023-04-25T23:19:55.696330355Z + [[ -n 1G ]]
2023-04-25T23:19:55.699138683Z [init] 2023-04-25 23:19:55+00:00 Setting initial memory to 1G and max to 1G
2023-04-25T23:19:55.699237447Z + log 'Setting initial memory to 1G and max to 1G'
2023-04-25T23:19:55.699286909Z + local oldState
2023-04-25T23:19:55.699292700Z ++ shopt -po xtrace
2023-04-25T23:19:55.699296427Z + oldState='set -o xtrace'
2023-04-25T23:19:55.699299883Z + shopt -u -o xtrace
2023-04-25T23:19:55.699309852Z + [[ -n 1G ]]
2023-04-25T23:19:55.699313688Z + JVM_OPTS='-Xms1G -Dlog4j2.formatMsgNoLookups=true  -DIReallyKnowWhatIAmDoingISwear'
2023-04-25T23:19:55.699317255Z + [[ -n 1G ]]
2023-04-25T23:19:55.699320691Z + JVM_OPTS='-Xmx1G -Xms1G -Dlog4j2.formatMsgNoLookups=true  -DIReallyKnowWhatIAmDoingISwear'
2023-04-25T23:19:55.699324358Z + mcServerRunnerArgs=(--stop-duration "${STOP_DURATION:-60}s" --named-pipe "${CONSOLE_IN_NAMED_PIPE:-/tmp/minecraft-console-in}")
2023-04-25T23:19:55.699327554Z + [[ -n '' ]]
2023-04-25T23:19:55.699330009Z + [[ BUKKIT == \C\U\R\S\E\F\O\R\G\E ]]
2023-04-25T23:19:55.699332744Z + [[ BUKKIT == \C\U\R\S\E\F\O\R\G\E ]]
2023-04-25T23:19:55.699335429Z + [[ craftbukkit_server-1.7.10-R0.1-20140808.005431-8.jar =~ run.sh ]]
2023-04-25T23:19:55.699337974Z + '[' -f /data/bootstrap.txt ']'
2023-04-25T23:19:55.699341691Z + log 'Starting the Minecraft server...'
2023-04-25T23:19:55.699345057Z + local oldState
2023-04-25T23:19:55.699541973Z ++ shopt -po xtrace
2023-04-25T23:19:55.699796388Z + oldState='set -o xtrace'
2023-04-25T23:19:55.699808830Z + shopt -u -o xtrace
2023-04-25T23:19:55.714609103Z [init] 2023-04-25 23:19:55+00:00 Starting the Minecraft server...
2023-04-25T23:19:55.715003006Z + finalArgs=($JVM_XX_OPTS $JVM_OPTS $expandedDOpts -jar "$SERVER" "$@" $EXTRA_ARGS)
2023-04-25T23:19:55.715129211Z + isTrue false
2023-04-25T23:19:55.715141624Z + case "${1,,}" in
2023-04-25T23:19:55.715151232Z + return 1
2023-04-25T23:19:55.715232694Z + isTrue false
2023-04-25T23:19:55.715240839Z + case "${1,,}" in
2023-04-25T23:19:55.715244647Z + return 1
2023-04-25T23:19:55.715248003Z + isTrue false
2023-04-25T23:19:55.715250968Z + case "${1,,}" in
2023-04-25T23:19:55.715254104Z + return 1
2023-04-25T23:19:55.715257060Z + exec mc-server-runner --stop-duration 60s --named-pipe /tmp/minecraft-console-in java -Xmx1G -Xms1G -Dlog4j2.formatMsgNoLookups=true -DIReallyKnowWhatIAmDoingISwear -jar craftbukkit_server-1.7.10-R0.1-20140808.005431-8.jar
2023-04-25T23:19:55.875833765Z Loading libraries, please wait...
2023-04-25T23:20:00.505483772Z [23:20:00 INFO]: Starting minecraft server version 1.7.10
2023-04-25T23:20:00.505765105Z [23:20:00 INFO]: Loading properties
2023-04-25T23:20:00.531362809Z [23:20:00 INFO]: Default game type: SURVIVAL
2023-04-25T23:20:00.532426990Z [23:20:00 INFO]: Generating keypair
2023-04-25T23:20:00.684239560Z [23:20:00 INFO]: Starting Minecraft server on *:25565
2023-04-25T23:20:00.861275106Z [23:20:00 INFO]: This server is running CraftBukkit version git-Bukkit-1.7.9-R0.2-20-g0b2ed13-b3108jnks (MC: 1.7.10) (Implementing API version 1.7.10-R0.1-SNAPSHOT)
2023-04-25T23:20:00.938710292Z [23:20:00 WARN]: Failed to load user banlist: 
2023-04-25T23:20:00.938790802Z java.io.FileNotFoundException: banned-players.json (No such file or directory)
2023-04-25T23:20:00.938843270Z 	at java.io.FileInputStream.open0(Native Method) ~[?:1.8.0_312]
2023-04-25T23:20:00.938847548Z 	at java.io.FileInputStream.open(FileInputStream.java:195) ~[?:1.8.0_312]
2023-04-25T23:20:00.938851225Z 	at java.io.FileInputStream.<init>(FileInputStream.java:138) ~[?:1.8.0_312]
2023-04-25T23:20:00.938855032Z 	at net.minecraft.util.com.google.common.io.Files.newReader(Files.java:86) ~[craftbukkit_server-1.7.10-R0.1-20140808.005431-8.jar:git-Bukkit-1.7.9-R0.2-20-g0b2ed13-b3108jnks]
2023-04-25T23:20:00.938858859Z 	at net.minecraft.server.v1_7_R4.JsonList.load(JsonList.java:147) ~[craftbukkit_server-1.7.10-R0.1-20140808.005431-8.jar:git-Bukkit-1.7.9-R0.2-20-g0b2ed13-b3108jnks]
2023-04-25T23:20:00.938862356Z 	at net.minecraft.server.v1_7_R4.DedicatedPlayerList.y(SourceFile:99) [craftbukkit_server-1.7.10-R0.1-20140808.005431-8.jar:git-Bukkit-1.7.9-R0.2-20-g0b2ed13-b3108jnks]
2023-04-25T23:20:00.938865832Z 	at net.minecraft.server.v1_7_R4.DedicatedPlayerList.<init>(SourceFile:25) [craftbukkit_server-1.7.10-R0.1-20140808.005431-8.jar:git-Bukkit-1.7.9-R0.2-20-g0b2ed13-b3108jnks]
2023-04-25T23:20:00.938869449Z 	at net.minecraft.server.v1_7_R4.DedicatedServer.init(DedicatedServer.java:133) [craftbukkit_server-1.7.10-R0.1-20140808.005431-8.jar:git-Bukkit-1.7.9-R0.2-20-g0b2ed13-b3108jnks]
2023-04-25T23:20:00.938872725Z 	at net.minecraft.server.v1_7_R4.MinecraftServer.run(MinecraftServer.java:436) [craftbukkit_server-1.7.10-R0.1-20140808.005431-8.jar:git-Bukkit-1.7.9-R0.2-20-g0b2ed13-b3108jnks]
2023-04-25T23:20:00.938876111Z 	at net.minecraft.server.v1_7_R4.ThreadServerApplication.run(SourceFile:628) [craftbukkit_server-1.7.10-R0.1-20140808.005431-8.jar:git-Bukkit-1.7.9-R0.2-20-g0b2ed13-b3108jnks]
2023-04-25T23:20:00.945377845Z [23:20:00 WARN]: Failed to load ip banlist: 
2023-04-25T23:20:00.945415004Z java.io.FileNotFoundException: banned-ips.json (No such file or directory)
2023-04-25T23:20:00.945419532Z 	at java.io.FileInputStream.open0(Native Method) ~[?:1.8.0_312]
2023-04-25T23:20:00.945424191Z 	at java.io.FileInputStream.open(FileInputStream.java:195) ~[?:1.8.0_312]
2023-04-25T23:20:00.945427778Z 	at java.io.FileInputStream.<init>(FileInputStream.java:138) ~[?:1.8.0_312]
2023-04-25T23:20:00.945431705Z 	at net.minecraft.util.com.google.common.io.Files.newReader(Files.java:86) ~[craftbukkit_server-1.7.10-R0.1-20140808.005431-8.jar:git-Bukkit-1.7.9-R0.2-20-g0b2ed13-b3108jnks]
2023-04-25T23:20:00.945436003Z 	at net.minecraft.server.v1_7_R4.JsonList.load(JsonList.java:147) ~[craftbukkit_server-1.7.10-R0.1-20140808.005431-8.jar:git-Bukkit-1.7.9-R0.2-20-g0b2ed13-b3108jnks]
2023-04-25T23:20:00.945439880Z 	at net.minecraft.server.v1_7_R4.DedicatedPlayerList.x(SourceFile:91) [craftbukkit_server-1.7.10-R0.1-20140808.005431-8.jar:git-Bukkit-1.7.9-R0.2-20-g0b2ed13-b3108jnks]
2023-04-25T23:20:00.945484314Z 	at net.minecraft.server.v1_7_R4.DedicatedPlayerList.<init>(SourceFile:27) [craftbukkit_server-1.7.10-R0.1-20140808.005431-8.jar:git-Bukkit-1.7.9-R0.2-20-g0b2ed13-b3108jnks]
2023-04-25T23:20:00.945489021Z 	at net.minecraft.server.v1_7_R4.DedicatedServer.init(DedicatedServer.java:133) [craftbukkit_server-1.7.10-R0.1-20140808.005431-8.jar:git-Bukkit-1.7.9-R0.2-20-g0b2ed13-b3108jnks]
2023-04-25T23:20:00.945492809Z 	at net.minecraft.server.v1_7_R4.MinecraftServer.run(MinecraftServer.java:436) [craftbukkit_server-1.7.10-R0.1-20140808.005431-8.jar:git-Bukkit-1.7.9-R0.2-20-g0b2ed13-b3108jnks]
2023-04-25T23:20:00.945505312Z 	at net.minecraft.server.v1_7_R4.ThreadServerApplication.run(SourceFile:628) [craftbukkit_server-1.7.10-R0.1-20140808.005431-8.jar:git-Bukkit-1.7.9-R0.2-20-g0b2ed13-b3108jnks]
2023-04-25T23:20:00.946610931Z [23:20:00 WARN]: Failed to load operators list: 
2023-04-25T23:20:00.946657046Z java.io.FileNotFoundException: ops.json (No such file or directory)
2023-04-25T23:20:00.946661124Z 	at java.io.FileInputStream.open0(Native Method) ~[?:1.8.0_312]
2023-04-25T23:20:00.946664841Z 	at java.io.FileInputStream.open(FileInputStream.java:195) ~[?:1.8.0_312]
2023-04-25T23:20:00.946668157Z 	at java.io.FileInputStream.<init>(FileInputStream.java:138) ~[?:1.8.0_312]
2023-04-25T23:20:00.946671483Z 	at net.minecraft.util.com.google.common.io.Files.newReader(Files.java:86) ~[craftbukkit_server-1.7.10-R0.1-20140808.005431-8.jar:git-Bukkit-1.7.9-R0.2-20-g0b2ed13-b3108jnks]
2023-04-25T23:20:00.946674830Z 	at net.minecraft.server.v1_7_R4.JsonList.load(JsonList.java:147) ~[craftbukkit_server-1.7.10-R0.1-20140808.005431-8.jar:git-Bukkit-1.7.9-R0.2-20-g0b2ed13-b3108jnks]
2023-04-25T23:20:00.946678136Z 	at net.minecraft.server.v1_7_R4.DedicatedPlayerList.z(SourceFile:107) [craftbukkit_server-1.7.10-R0.1-20140808.005431-8.jar:git-Bukkit-1.7.9-R0.2-20-g0b2ed13-b3108jnks]
2023-04-25T23:20:00.946681492Z 	at net.minecraft.server.v1_7_R4.DedicatedPlayerList.<init>(SourceFile:29) [craftbukkit_server-1.7.10-R0.1-20140808.005431-8.jar:git-Bukkit-1.7.9-R0.2-20-g0b2ed13-b3108jnks]
2023-04-25T23:20:00.946695888Z 	at net.minecraft.server.v1_7_R4.DedicatedServer.init(DedicatedServer.java:133) [craftbukkit_server-1.7.10-R0.1-20140808.005431-8.jar:git-Bukkit-1.7.9-R0.2-20-g0b2ed13-b3108jnks]
2023-04-25T23:20:00.946699194Z 	at net.minecraft.server.v1_7_R4.MinecraftServer.run(MinecraftServer.java:436) [craftbukkit_server-1.7.10-R0.1-20140808.005431-8.jar:git-Bukkit-1.7.9-R0.2-20-g0b2ed13-b3108jnks]
2023-04-25T23:20:00.946702471Z 	at net.minecraft.server.v1_7_R4.ThreadServerApplication.run(SourceFile:628) [craftbukkit_server-1.7.10-R0.1-20140808.005431-8.jar:git-Bukkit-1.7.9-R0.2-20-g0b2ed13-b3108jnks]
2023-04-25T23:20:00.947104069Z [23:20:00 WARN]: Failed to load white-list: 
2023-04-25T23:20:00.947135447Z java.io.FileNotFoundException: whitelist.json (No such file or directory)
2023-04-25T23:20:00.947139604Z 	at java.io.FileInputStream.open0(Native Method) ~[?:1.8.0_312]
2023-04-25T23:20:00.947143251Z 	at java.io.FileInputStream.open(FileInputStream.java:195) ~[?:1.8.0_312]
2023-04-25T23:20:00.947162067Z 	at java.io.FileInputStream.<init>(FileInputStream.java:138) ~[?:1.8.0_312]
2023-04-25T23:20:00.947166064Z 	at net.minecraft.util.com.google.common.io.Files.newReader(Files.java:86) ~[craftbukkit_server-1.7.10-R0.1-20140808.005431-8.jar:git-Bukkit-1.7.9-R0.2-20-g0b2ed13-b3108jnks]
2023-04-25T23:20:00.947169771Z 	at net.minecraft.server.v1_7_R4.JsonList.load(JsonList.java:147) ~[craftbukkit_server-1.7.10-R0.1-20140808.005431-8.jar:git-Bukkit-1.7.9-R0.2-20-g0b2ed13-b3108jnks]
2023-04-25T23:20:00.947173568Z 	at net.minecraft.server.v1_7_R4.DedicatedPlayerList.B(SourceFile:123) [craftbukkit_server-1.7.10-R0.1-20140808.005431-8.jar:git-Bukkit-1.7.9-R0.2-20-g0b2ed13-b3108jnks]
2023-04-25T23:20:00.947177345Z 	at net.minecraft.server.v1_7_R4.DedicatedPlayerList.<init>(SourceFile:30) [craftbukkit_server-1.7.10-R0.1-20140808.005431-8.jar:git-Bukkit-1.7.9-R0.2-20-g0b2ed13-b3108jnks]
2023-04-25T23:20:00.947181062Z 	at net.minecraft.server.v1_7_R4.DedicatedServer.init(DedicatedServer.java:133) [craftbukkit_server-1.7.10-R0.1-20140808.005431-8.jar:git-Bukkit-1.7.9-R0.2-20-g0b2ed13-b3108jnks]
2023-04-25T23:20:00.947184699Z 	at net.minecraft.server.v1_7_R4.MinecraftServer.run(MinecraftServer.java:436) [craftbukkit_server-1.7.10-R0.1-20140808.005431-8.jar:git-Bukkit-1.7.9-R0.2-20-g0b2ed13-b3108jnks]
2023-04-25T23:20:00.947188436Z 	at net.minecraft.server.v1_7_R4.ThreadServerApplication.run(SourceFile:628) [craftbukkit_server-1.7.10-R0.1-20140808.005431-8.jar:git-Bukkit-1.7.9-R0.2-20-g0b2ed13-b3108jnks]
2023-04-25T23:20:00.955774008Z [23:20:00 INFO]: Preparing level "world"
2023-04-25T23:20:01.162558767Z [23:20:01 WARN]: Exception in thread "Thread-6" 
2023-04-25T23:20:01.163251106Z [23:20:01 WARN]: org.bukkit.craftbukkit.libs.com.google.gson.JsonSyntaxException: java.lang.IllegalStateException: Expected BEGIN_OBJECT but was STRING at line 1 column 7
2023-04-25T23:20:01.163643907Z [23:20:01 WARN]: 	at org.bukkit.craftbukkit.libs.com.google.gson.internal.bind.ReflectiveTypeAdapterFactory$Adapter.read(ReflectiveTypeAdapterFactory.java:180)
2023-04-25T23:20:01.163989941Z [23:20:01 WARN]: 	at org.bukkit.craftbukkit.libs.com.google.gson.Gson.fromJson(Gson.java:755)
2023-04-25T23:20:01.164407128Z [23:20:01 WARN]: 	at org.bukkit.craftbukkit.libs.com.google.gson.Gson.fromJson(Gson.java:694)
2023-04-25T23:20:01.164742332Z [23:20:01 WARN]: 	at org.bukkit.craftbukkit.v1_7_R4.updater.BukkitDLUpdaterService.fetchArtifact(BukkitDLUpdaterService.java:53)
2023-04-25T23:20:01.165083627Z [23:20:01 WARN]: 	at org.bukkit.craftbukkit.v1_7_R4.updater.BukkitDLUpdaterService.getArtifact(BukkitDLUpdaterService.java:30)
2023-04-25T23:20:01.165467151Z [23:20:01 WARN]: 	at org.bukkit.craftbukkit.v1_7_R4.updater.AutoUpdater$1.run(AutoUpdater.java:77)
2023-04-25T23:20:01.165887524Z [23:20:01 WARN]: Caused by: java.lang.IllegalStateException: Expected BEGIN_OBJECT but was STRING at line 1 column 7
2023-04-25T23:20:01.166295082Z [23:20:01 WARN]: 	at org.bukkit.craftbukkit.libs.com.google.gson.stream.JsonReader.expect(JsonReader.java:339)
2023-04-25T23:20:01.166659120Z [23:20:01 WARN]: 	at org.bukkit.craftbukkit.libs.com.google.gson.stream.JsonReader.beginObject(JsonReader.java:322)
2023-04-25T23:20:01.166996468Z [23:20:01 WARN]: 	at org.bukkit.craftbukkit.libs.com.google.gson.internal.bind.ReflectiveTypeAdapterFactory$Adapter.read(ReflectiveTypeAdapterFactory.java:168)
2023-04-25T23:20:01.167452587Z [23:20:01 WARN]: 	... 5 more
2023-04-25T23:20:01.223456987Z [23:20:01 INFO]: Preparing start region for level 0 (Seed: -8326298013595310485)
2023-04-25T23:20:02.245170899Z [23:20:02 INFO]: Preparing spawn area: 11%
2023-04-25T23:20:03.255592156Z [23:20:03 INFO]: Preparing spawn area: 22%
2023-04-25T23:20:04.259206311Z [23:20:04 INFO]: Preparing spawn area: 37%
2023-04-25T23:20:05.261518302Z [23:20:05 INFO]: Preparing spawn area: 62%
2023-04-25T23:20:06.262266802Z [23:20:06 INFO]: Preparing spawn area: 86%
2023-04-25T23:20:06.911362703Z [23:20:06 INFO]: Preparing start region for level 1 (Seed: -8326298013595310485)
2023-04-25T23:20:07.912596226Z [23:20:07 INFO]: Preparing spawn area: 37%
2023-04-25T23:20:08.932296859Z [23:20:08 INFO]: Preparing spawn area: 69%
2023-04-25T23:20:09.601539526Z [23:20:09 INFO]: Preparing start region for level 2 (Seed: -8326298013595310485)
2023-04-25T23:20:10.603362887Z [23:20:10 INFO]: Preparing spawn area: 84%
2023-04-25T23:20:10.757550899Z [23:20:10 INFO]: Done (9.805s)! For help, type "help" or "?"
2023-04-25T23:20:10.760503376Z [23:20:10 INFO]: Starting remote control listener
2023-04-25T23:20:10.764239360Z [23:20:10 INFO]: RCON running on 0.0.0.0:25575
2023-04-25T23:20:15.714461416Z [23:20:15 INFO]: UUID of player ######## is ########
2023-04-25T23:20:15.776565210Z [23:20:15 INFO]: ########[/########:60911] logged in with entity id 2360 at ([world] -8.5, 64.0, 142.5)
2023-04-25T23:20:23.267410362Z [23:20:23 INFO]: <########> Log4Shell Test
2023-04-25T23:20:34.864669770Z [23:20:34 INFO]: <########> ${jndi:ldap://log4shell.huntress.com:1389/########-####-####-####-############}
2023-04-25T23:20:54.367945099Z [23:20:54 INFO]: <########> still vulnerable
2023-04-25T23:21:11.587373826Z 2023/04/25 23:21:11 Stopping with rcon-cli
2023-04-25T23:21:11.600336318Z [23:21:11 INFO]: Rcon connection from: /127.0.0.1
2023-04-25T23:21:11.619296114Z [23:21:11 INFO]: Rcon: Stopping the server..
2023-04-25T23:21:11.622772154Z 2023-04-25T23:21:11.622Z	INFO	mc-server-runner	Waiting for completion...
2023-04-25T23:21:11.670882890Z [23:21:11 INFO]: Stopping server
2023-04-25T23:21:11.671681877Z [23:21:11 INFO]: Saving players
2023-04-25T23:21:11.679054082Z [23:21:11 INFO]: ######## lost connection: Server closed
2023-04-25T23:21:11.700903768Z [23:21:11 INFO]: ######## left the game.
2023-04-25T23:21:11.701104732Z [23:21:11 INFO]: Saving worlds
2023-04-25T23:21:11.701215298Z [23:21:11 INFO]: Saving chunks for level 'world'/Overworld
2023-04-25T23:21:12.518627633Z [23:21:12 INFO]: Saving chunks for level 'world_nether'/Nether
2023-04-25T23:21:12.518672045Z [23:21:12 INFO]: Saving chunks for level 'world_the_end'/The End
2023-04-25T23:21:12.521746770Z [23:21:12 INFO]: Stopping server
2023-04-25T23:21:12.867950326Z 2023-04-25T23:21:12.867Z	INFO	mc-server-runner	Done
```

--- a/Dockerfile
+++ b/Dockerfile
@@ -70,6 +70,8 @@ COPY --chmod=644 files/log4j2.xml /image/log4j2.xml
 COPY --chmod=644 files/cf-exclude-include.json /image/cf-exclude-include.json
 COPY --chmod=755 files/auto /auto
 
+RUN curl -fsSL -o /image/Log4jPatcher.jar https://github.com/CreeperHost/Log4jPatcher/releases/download/v1.0.1/Log4jPatcher-1.0.1.jar
+
 RUN dos2unix /start* /auto/*
 
 ENTRYPOINT [ "/start" ]

--- a/scripts/start-finalExec
+++ b/scripts/start-finalExec
@@ -77,6 +77,16 @@ if ${useFallbackJvmFlag}; then
   JVM_OPTS="-Dlog4j2.formatMsgNoLookups=true ${JVM_OPTS}"
 fi
 
+if versionLessThan 1.7; then
+  : # No patch required here.
+elif versionLessThan 1.18.1; then
+  if isTrue ${SKIP_LOG4J_PATCHER:-false}; then
+    log "Skipping Log4jPatcher, make sure you are not affected"
+  else
+    JVM_OPTS="-javaagent:/image/Log4jPatcher.jar ${JVM_OPTS}"
+  fi
+fi
+
 if isTrue ${ENABLE_ROLLING_LOGS:-false}; then
   if ! ${canUseRollingLogs}; then
     log "ERROR: Using rolling logs is currently not possible in the selected version due to CVE-2021-44228"

